ipc/rpc/src/binary.rs

```diff
@@ -706,14 +705,21 @@ pub fn serialize<T: BinaryConvertable>(t: &T) -> Result<Vec<u8>, BinaryError> {
 #[macro_export]
 macro_rules! binary_fixed_size {
 	($target_ty: ty) => {
-		impl BinaryConvertable for $target_ty {
+		impl BinaryConvertable for $target_ty where $target_ty: Copy {
 			fn from_bytes(bytes: &[u8], _length_stack: &mut ::std::collections::VecDeque<usize>) -> Result<Self, BinaryConvertError> {
-				match bytes.len().cmp(&::std::mem::size_of::<$target_ty>()) {
+				let size = ::std::mem::size_of::<$target_ty>();
+				match bytes.len().cmp(&size) {
 					::std::cmp::Ordering::Equal => (),
-					_ => return Err(BinaryConvertError::size(::std::mem::size_of::<$target_ty>(), bytes.len())),
+					_ => return Err(BinaryConvertError::size(size, bytes.len())),
 				};
-				let mut res: Self = unsafe { ::std::mem::uninitialized() };
```

Comment:
I don't think it's strictly safer but in general it's better practice than using `uninitialized`. LLVM can optimize out the zero-ing so I doubt we'd see a performance penalty. `uninitialized` imo should be used only for FFI and dirty tricks.